doc-surrealist_versioned_docs/version-2.x/tutorials/How-to-define-a-schema-in-Surrealist.mdx

```diff
@@ -0,0 +1,57 @@
+
+# Defining A Schema In Surrealist
+
+Surrealist allows you to design your schemas both visually in the [designer view](/docs/surrealist/getting-started/designing-the-database-schema) and with [queries in the query view](/docs/surrealist/getting-started/sending-queries), making it easy to create and manage your tables and relationships. 
+
+In this guide, you will walk through defining a schema in Surrealist.
+
+## What is a schema?
+
+A schema is a blueprint of your database that defines the structure of your tables, the relationships between them, and the constraints that apply to your data. 
+
+## Creating a new schema in the designer view
+
+In Surrealist, you can create a new schema in the designer view by defining your tables and their columns, setting primary and foreign keys, and establishing relationships between your tables.
+
+To create a new schema in Surrealist, follow these steps:
+
+1. Open the designer view by clicking the designer icon in the sidebar.
+
+2. Click the `+` button to create a new table or a new relation between tables.
+
+3. Enter the table name and click the `create` button.
+
+After your table is created, you can add columns by double-tapping the table name. You can then define permissions,changefeeds, fields, and constraints for the table. You can also toggle the table type depending on your requirements.
+
+:::note
+___NOTE:___ By default, the table is schemaless, but you can enforce a schema by toggling the schema button.
+::: 
+
+## Creating a new schema with the query view
+
+You can also create a new schema in Surrealist using the query view by sending a [`DEFINE TABLE`](/docs/surrealdb/surrealql/statements/define/table) query.
+
+For example, to define a new `SCHEMAFULL` table called `users` with columns `firstname`, `lastname`, and `email`, you can send the following query:
+
+```surql
+-- Create schemafull user table.
+DEFINE TABLE user SCHEMAFULL;
+
+-- Define some fields.
+DEFINE FIELD firstName ON TABLE user TYPE string;
+DEFINE FIELD lastName ON TABLE user TYPE string;
+DEFINE FIELD email ON TABLE user TYPE string
+  ASSERT string::is::email($value);
+DEFINE INDEX userEmailIndex ON TABLE user COLUMNS email UNIQUE;
+```
+
+<SurrealistMini url="https://surrealist.app/mini?setup=--+Create+schemafull+user+table.%0ADEFINE+TABLE+user+SCHEMAFULL%3B%0A%0A--+Define+some+fields.%0ADEFINE+FIELD+firstName+ON+TABLE+user+TYPE+string%3B%0ADEFINE+FIELD+lastName+ON+TABLE+user+TYPE+string%3B%0ADEFINE+FIELD+email+ON+TABLE+user+TYPE+string%0A++ASSERT+string%3A%3Ais%3A%3Aemail%28%24value%29%3B%0ADEFINE+INDEX+userEmailIndex+ON+TABLE+user+COLUMNS+email+UNIQUE%3B&query=%2F%2F+Get+information+for+Table%0AINFO+FOR+TABLE+user%3B%0A%0A%2F%2F+Add+a+user+with+all+required+fields+%0ACREATE+user+CONTENT+%7B%0A++++firstName%3A+%27Tobie%27%2C%0A++++lastName%3A+%27Hitchcock%27%2C%0A++++email%3A+%27Tobie.Hitchcock%40surrealdb.com%27%2C%0A%7D%3B%0A%0A%2F%2F+Query+the+data%0ASELECT+*+FROM+user%3B"/>
```

Comment:
Also just an idea that you can ignore: I like pointing out that SCHEMAFULL means that you now have to define each and every field before you can use it, so if that is annoying then DEFINE FIELD per field is a less strict option.

```
// With a SCHEMAFULL table, each field has to be defined before it can be set.
// If DEFINE TABLE is commented out, the DEFINE FIELD statements below
// would apply a schema only on each field but the table itself would
// remain freeform
DEFINE TABLE user SCHEMAFULL;

DEFINE FIELD firstName ON TABLE user TYPE string;
DEFINE FIELD lastName ON TABLE user TYPE string;
DEFINE FIELD email ON TABLE user TYPE string
  ASSERT string::is::email($value);
DEFINE INDEX userEmailIndex ON TABLE user COLUMNS email UNIQUE;

// Get information for Table
INFO FOR TABLE user;

// Add a user with all required fields 
CREATE user CONTENT {
    firstName: 'Tobie',
    lastName: 'Hitchcock',
    email: 'Tobie.Hitchcock@surrealdb.com',
};

// Query the data
SELECT * FROM user;
```